include/wabt/ir.h

```diff
@@ -255,12 +255,24 @@ struct FuncSignature {
   TypeVector param_types;
   TypeVector result_types;
 
-  // Some types can have names, for example (ref $foo) has type $foo.
-  // So to use this type we need to translate its name into
-  // a proper index from the module type section.
-  // This is the mapping from parameter/result index to its name.
-  std::unordered_map<uint32_t, std::string> param_type_names;
-  std::unordered_map<uint32_t, std::string> result_type_names;
+  // Reference types can have names, for example (ref $foo) represents
+  // a type which name is $foo. These types are translated to their
+  // corresponding index after the parsing is completed. The position
+  // and names of these reference types are stored in a ReferenceVars
+  // vector. The names are stored as variables, because the error
+  // message construction requires a location when a name is not found.
+  struct ReferenceVar {
+    ReferenceVar(uint32_t index, Var var)
+      : index(index), var(var) {}
+
+    uint32_t index;
```

Comment:
what is the meaning of this index?

Reply:
Location of the reference.

Example: `(param i32 (ref $a) i64 (ref 0) f32 (ref $b))`, then two `ReferenceVar` is created: `(1, $a)` and `(5, $b)`. The index is 1 and 5.

Shall I add a comment?

Reply:
So this whole class is only needed/used during parsing?  Should we add a TODO to move this into the parser itself in the future?